### ISSUE TITLE
Failing tests on arm64

### ISSUE THREAD

**Issue by bobrik** [VictoriaMetrics/VictoriaMetrics#1074]
**Describe the bug**

I see a few failures when trying to test on arm64 (Linux VM on an M1 based mac):

```
=== RUN   TestRollupFuncsNoWindow/deriv
    rollup_test.go:959: unexpected value at values[1]; got -2879.310345; want -2879.310345
        values=
        [0 -2879.3103448275865 558.0608793686582 422.84569138276555 3.469446951953614e-13]
        valuesExpected=
        [0 -2879.310344827587 558.0608793686595 422.84569138276544 NaN]
```

```
=== RUN   TestRollupNewRollupFuncSuccess
    rollup_test.go:451: unexpected value; got -266.8586023140607; want -266.85860231406065
```

```
=== RUN   TestRollupPredictLinear
    rollup_test.go:361: unexpected value; got 17.039502356142; want 17.03950235614201
```

That's with Go 1.15.

**To Reproduce**

Run tests on arm64.

**Expected behavior**

Tests pass.

**Version**

1.52.0

**Comment by bobrik** [comment]
https://github.com/golang/snappy/commit/f81760ec4c9208e6e6866cf18e1888544fd2dbc9 needs to be picked up to avoid the following failure on Go 1.16:

```
goroutine 1 [running, locked to thread]:
runtime.throw(0x131256e, 0x5)
        runtime/panic.go:1117 +0x54 fp=0x4000539560 sp=0x4000539530 pc=0x465a4
runtime.sigpanic()
        runtime/signal_unix.go:741 +0x230 fp=0x40005395a0 sp=0x4000539560 pc=0x5ddc0
github.com/golang/snappy.encodeBlock(0x4000a96014, 0x12ac8, 0x12ac8, 0x4000a86000, 0x89, 0x10000, 0x4247df4f)
        github.com/golang/snappy@v0.0.2/encode_arm64.s:666 +0x360 fp=0x4000541640 sp=0x40005395b0 pc=0x4780b0
github.com/golang/snappy.Encode(0x4000a96012, 0x12aca, 0x12aca, 0x0, 0x0, 0x0, 0x40003ca060, 0x4000236600, 0x89)
        github.com/golang/snappy@v0.0.2/encode.go:39 +0x17c fp=0x4000541690 sp=0x4000541640 pc=0x476d3c
github.com/golang/snappy.(*Writer).write(0x40003ca060, 0x4000a86000, 0x0, 0x10000, 0x200, 0x4000236600, 0x40003ca0c0)
        github.com/golang/snappy@v0.0.2/encode.go:233 +0x130 fp=0x4000541730 sp=0x4000541690 pc=0x477250
github.com/golang/snappy.(*Writer).Flush(0x40003ca060, 0x117b2a0, 0x40000b2180)
        github.com/golang/snappy@v0.0.2/encode.go:276 +0x60 fp=0x4000541780 sp=0x4000541730 pc=0x477610
github.com/golang/snappy.(*Writer).Close(...)
        github.com/golang/snappy@v0.0.2/encode.go:283
github.com/cortexproject/cortex/pkg/chunk.(*Chunk).Encode(0x40000b2180, 0x0, 0x0)
        github.com/cortexproject/cortex@v1.4.1-0.20201022071705-85942c5703cf/pkg/chunk/chunk.go:205 +0x1a8 fp=0x4000541830 sp=0x4000541780 pc=0x837378
github.com/grafana/loki/pkg/storage.newChunk(0x131b08a, 0xb, 0x205f240, 0x3, 0x3, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        github.com/grafana/loki/pkg/storage/util_test.go:113 +0x578 fp=0x4000541ab0 sp=0x4000541830 pc=0xf15d08
github.com/grafana/loki/pkg/storage.newMockChunkStore(0x20537a0, 0x4, 0x4, 0x4c4b40)
        github.com/grafana/loki/pkg/storage/util_test.go:163 +0xd8 fp=0x4000541c50 sp=0x4000541ab0 pc=0xf16298
github.com/grafana/loki/pkg/storage.init()
        github.com/grafana/loki/pkg/storage/util_test.go:328 +0x92c fp=0x4000541cb0 sp=0x4000541c50 pc=0xf1b83c
runtime.doInit(0x1ffa100)
        runtime/proc.go:6265 +0xdc fp=0x4000541e10 sp=0x4000541cb0 pc=0x56a8c
runtime.doInit(0x1fe6b60)
        runtime/proc.go:6242 +0x60 fp=0x4000541f70 sp=0x4000541e10 pc=0x56a10
runtime.main()
        runtime/proc.go:208 +0x214 fp=0x4000541fd0 sp=0x4000541f70 pc=0x48a94
runtime.goexit()
        runtime/asm_arm64.s:1130 +0x4 fp=0x4000541fd0 sp=0x4000541fd0 pc=0x7d0d4
```

**Comment by valyala** [comment]
@bobrik , thanks for the pointer! The https://github.com/golang/snappy/commit/f81760ec4c9208e6e6866cf18e1888544fd2dbc9 is included in the commit 4281c5ed147e118bbfa487207ad80240e95c9e88 , which will be included in the next release of VictoriaMetrics.

**Comment by valyala** [comment]
The bugfix has been included in [v1.55.0](https://github.com/VictoriaMetrics/VictoriaMetrics/releases). Closing the issue as fixed.
